### PR TITLE
Disallow bare pytest.raises

### PR DESCRIPTION
**Pull request by DylanBrdt**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

https://github.com/pandas-dev/pandas/issues/30999

### CODE REVIEW

File: pandas/tests/io/test_html.py [hunk 1]
```diff
@@ -276,14 +276,14 @@ def test_file_like(self):
 
     @tm.network
     def test_bad_url_protocol(self):
-        with pytest.raises(URLError):
+        with pytest.raises(URLError, match=msg):
```
Comment (jbrockmendel) [inline pandas/tests/io/test_html.py]: where is msg defined?

**Comment by simonjayhawkins** [comment]
Thanks @DylanBrdt for the PR. In future you should create a new branch on your local fork when making a PR request. see https://dev.pandas.io/docs/development/contributing.html#creating-a-branch

File: pandas/tests/io/test_stata.py [hunk 2]
```diff
@@ -1374,7 +1374,7 @@ def test_unsupported_datetype(self):
                 "dates": dates,
             }
         )
-        with pytest.raises(NotImplementedError):
+        with pytest.raises(NotImplementedError, match=msg):
```
Comment (simonjayhawkins) [inline pandas/tests/io/test_stata.py]:
```suggestion
        with pytest.raises(
            NotImplementedError,
            match=re.escape("Data type datetime64[ns, Asia/Hong_Kong] not supported."),
        ):
```

also re needs to be imported.

**Comment by MarcoGorelli** [comment]
Hi @DylanBrdt - are you still interested in working on this?

**Comment by DylanBrdt** [comment]
@MarcoGorelli I appreciate your consideration of asking me. Unfortunately, I am still learning OSS and want to learn the basics of git before I attempt any more issues.
Thank you.

**Comment by simonjayhawkins** [comment]
OK. closing for now.

**Pull request closed without merging**